internal/controller/crossplane_controller.go

```diff
@@ -181,12 +185,15 @@ func (r *CrossplaneReconciler) setupClusterAccess(ctx context.Context, req ctrl.
 }
 
 func (r *CrossplaneReconciler) setupFluxKubeconfig(ctx context.Context, req ctrl.Request) (context.Context, error) {
-	tenantNamespace := libutils.StableRequestNamespace(req.Namespace)
+	tenantNamespace, err := libutils.StableMCPNamespace(req.Name, req.Namespace)
+	if err != nil {
+		return ctx, fmt.Errorf("failed to determine stable namespace for Crossplane instance: %w", err)
+	}
 
 	// Get MCP AccessRequest to use for Flux
 	mcpAccessRequest := &clustersv1alpha1.AccessRequest{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      libutils.StableRequestNameMCP(req.Name, controllerName),
+			Name:      clusteraccess.StableRequestName(controllerName, req),
 			Namespace: tenantNamespace,
 		},
 	}
```

Comment:
@Diaphteiros Could you review these lines please? Background: Flux needs to know where can find the Secret with the kubeconfig to the MCP. Therefore, I need read the `AccessRequest` again. 

I am using `libutils.StableMCPNamespace(req.Name, req.Namespace)` for determining the namespace and `clusteraccess.StableRequestName(controllerName, req)` for the name of the `AccessRequest` resource. Is this correct?